### PR TITLE
Add classPrefix support

### PR DESCRIPTION
**Pull request by darrarski**
### Short description 📝

This PR introduces changes that allows (optionally) defining "class prefix" for Xcode project. The "class prefix" is a prefix for class files Xcode generates when you create a project or class file. For example, it's used when Xcode generates source files from intent definition.

### How to test the changes locally 🧐

1. Set `classPrefix` argument in `Project.swift` manifest:
   ```swift
   import ProjectDescription

   let project = Project(
     name: "DemoApp",
     classPrefix: "TUIST",
     // ...
   )
   ```
2. Run `tuist generate` to generate the Xcode project.
3. Generated Xcode project should have the correct class prefix set, which can be verified in Xcode's inspector panel when the project is selected in project navigator:
   Screenshot 2024-06-25 at 17 02 48

### Contributor checklist ✅

- [x] The code has been linted using run `mise run lint-fix`
- [x] The change is tested via unit testing or acceptance testing, or both
- [x] The title of the PR is formulated in a way that is usable as a changelog entry
- [x] In case the PR introduces changes that affect users, the documentation has been updated

### Reviewer checklist ✅

- [ ] The code architecture and patterns are consistent with the rest of the codebase
- [ ] Reviewer has checked that, if needed, the PR includes the label `changelog:added`, `changelog:fixed`, or `changelog:changed`, and the title is usable as a changelog entry

### CODE REVIEW

**Comment by darrarski** [comment]
Changes in this PR depends on my [fork of `tuist/XcodeGraph`](https://github.com/darrarski/XcodeGraph/tree/add-class-prefix-to-project) repository. I've created another [PR in `tuist/XcodeGraph`](https://github.com/tuist/XcodeGraph/pull/14) that should be merged before this one. Once done, this PR should be updated to point to the `tuist/XcodeGraph` repository, instead of my fork.

I'm not sure what's the correct way of handling this kind of contribution. I need some guidance/help to make it right.

File: Package.resolved [hunk 1]
```diff
@@ -238,10 +238,10 @@
     {
       "identity" : "xcodegraph",
       "kind" : "remoteSourceControl",
-      "location" : "https://github.com/tuist/XcodeGraph.git",
+      "location" : "https://github.com/darrarski/XcodeGraph.git",
       "state" : {
-        "revision" : "0e44220403553ce7f20a79e49873b492302b4650",
-        "version" : "0.5.0"
+        "branch" : "add-class-prefix-to-project",
+        "revision" : "627830f87824cf7b1bc62298d86475d32a0ec1e2"
```
Comment (pepicrft) [inline Package.resolved:244]: You can revert these changes and point your PR to the version 0.7.0 or XcodeGraph :)

**Review by pepicrft — APPROVED** [review]
Thanks a lot for contributing this one @darrarski 👏🏼

**Comment by pepicrft** [comment]
@all-contributors add @darrarski for code

File: Package.resolved (hunk 1 above)
Comment (darrarski) [reply Package.resolved:244]: I've updated the dependency so it points to `tuist/XcodeGraph` version `.upToNextMajor(from: "0.7.0")`. Unfortunately, there is a lot of changes between `0.5.0` and `0.7.0`, causing many build errors in the base Tuist repo. Not sure if I'm able to resolve them all (I tried, but it touches a lot of files, which I'm not familiar with).

Could we hold on, until the bump to `XcodeGraph` `0.7.0` will be made, possibly in a separate PR, by someone more experienced in Tuist's codebase?

**Review by fortmarek — COMMENTED** [review]
Changes look good 🙌  I'd consider adding a new fixture `app_with_class_prefix` and a new acceptance test to ensure the project with this property builds.

You can create a new fixture by running `tuist init` in `fixtures/app_with_class_prefix` and updating the project to use `classPrefix`. You can then test the project is generated and can be built successfully in `GenerateAcceptanceTests`.

**Review by fortmarek — APPROVED** [review]
Thanks a lot for updating this PR 👏 lgtm
